From: Lance Wang <lzwang@suse.com>
Subject: Support yes/no style for KDUMP_CONTINUE_ON_ERROR
References: bsc#1083155
Upstream: tbd

The /etc/sysconfig/kdump in latest kdump initrd is generated by
kdumptool. So the value of KDUMP_CONTINUE_ON_ERROR is yes instead
of true. This patch make save_dump.sh support this value.

Signed-off-by: Lance Wang <lzwang@suse.com>
Acked-by: Petr Tesarik <ptesarik@suse.com>

---
 init/save_dump.sh |   22 +++++++++++++---------
 1 file changed, 13 insertions(+), 9 deletions(-)

--- a/init/save_dump.sh
+++ b/init/save_dump.sh
@@ -101,16 +101,20 @@ function continue_error()
 
     echo "Last command failed ($status)."
 
-    if ! [ "$KDUMP_CONTINUE_ON_ERROR" = "true" -o \
-                "$KDUMP_CONTINUE_ON_ERROR" = "TRUE" ] ; then
-        echo
-        echo "Something failed. You can try to debug that here."
-        echo "Type 'reboot -f' to reboot the system or 'exit' to"
-        echo "boot in a normal system that still is running in the"
-        echo "kdump environment with restricted memory!"
-        bash
-	return 1
+    if [ "$KDUMP_CONTINUE_ON_ERROR" = "yes" -o \
+         "$KDUMP_CONTINUE_ON_ERROR" = "YES" -o \
+         "$KDUMP_CONTINUE_ON_ERROR" = "true" -o \
+         "$KDUMP_CONTINUE_ON_ERROR" = "TRUE" ]; then
+        return 0
     fi
+
+    echo
+    echo "Something failed. You can try to debug that here."
+    echo "Type 'reboot -f' to reboot the system or 'exit' to"
+    echo "boot in a normal system that still is running in the"
+    echo "kdump environment with restricted memory!"
+    bash
+    return 1
 }
 
 function rw_fixup()
